presto-hive/src/test/java/com/facebook/presto/hive/HiveTestUtils.java

```diff
@@ -267,4 +269,28 @@ public static Slice longDecimal(String value)
     {
         return encodeScaledValue(new BigDecimal(value));
     }
+
+    public static Optional<String> getProperty(String name)
+    {
+        String systemPropertyValue = System.getProperty(name);
+        String environmentVariableValue = System.getenv(name);
+        if (systemPropertyValue == null) {
+            if (environmentVariableValue == null) {
+                return Optional.empty();
+            }
+            else {
+                return Optional.of(environmentVariableValue);
+            }
+        }
+        else {
+            if (environmentVariableValue != null && !systemPropertyValue.equals(environmentVariableValue)) {
```

Comment:
I'm glad you're adding this check. It will definitely prevent even more confusing situations.